### PR TITLE
Add C# 9 test cases for S3869, S1854, S3346, S3903, S4524, S3172

### PR DESCRIPTION
**Pull request by andrei-epure-sonarsource**
Related to #3668

### CODE REVIEW

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DeclareTypesInNamespacesTest.cs [hunk 1]
```diff
@@ -54,11 +54,17 @@ public void DeclareTypesInNamespaces_CS_Before8()
 
         [TestMethod]
         [TestCategory("Rule")]
-        public void DeclareTypesInNamespaces_VB()
-        {
+        public void DeclareTypesInNamespaces_CS_AfterCSharp9() =>
+            Verifier.VerifyAnalyzer(@"TestCases\DeclareTypesInNamespaces.AfterCSharp9.cs",
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DeclareTypesInNamespacesTest.cs:58]: Why not follow the same convention with `CSharp9`?

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeadStores.CSharp9.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+    nint two = 2; // Noncompliant
+    two = 3;
+    Foo(two);
+}
+
+void PatternMatch(object param)
+{
+    object a = param;
+    if (a is not null)
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeadStores.CSharp9.cs:36]: `Not` is not implemented on our side and I think we should add a case to show this. Maybe something similar with:
```
void PatternMatch2(int a, int b)
{
    if (b is not 5)
    {
        a = 1;
    }
    else if (b is 5)
    {
        a = 2; // Compliant - FN, the parameter value is overwritten on all possible paths
    }
}
```

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeadStores.CSharp9.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    {
+        a = null; // Compliant
+        Foo(a);
+    }
+
+    int i = 100;
+    if (i is not > 50 and < 200)
```
Comment (costin-zaharia-sonarsource) [inline sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/DeadStores.CSharp9.cs:43]: Same comment as above. We don't support the new patterns:
```
var c = 5;
    switch (c)
    {
        case < 5:
            c = 5; // Compliant, FN
            break;
        case >= 5:
            c = 56; // Compliant, FN
            break;
    }
}
```

**Review by costin-zaharia-sonarsource — APPROVED** [review]
LGTM! Please see my comments with missing test cases for DeadStores.

File: sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DeclareTypesInNamespacesTest.cs (hunk 1 above)
Comment (andrei-epure-sonarsource) [reply sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/DeclareTypesInNamespacesTest.cs:58]: to have the same pattern with the other files
`DeclareTypesInNamespaces_CS_Before8`
`DeclareTypesInNamespaces_CS_After8`

same for the test file name

**Pull request merged**